store/src/main/java/com/zimbra/cs/imap/ImapRequest.java

```diff
@@ -1023,14 +1024,15 @@ protected ImapPartSpecifier readPartSpecifier(boolean binary, boolean literals)
                 }
                 skipChar(')');
             } else if (sectionText.equals("MIME")) {
-                if (sectionPart.isEmpty()) {
+                if (sectionPart.length() == 0) {
```

Comment:
According to Java implementation of `isEmpty()`:
```java
    public boolean isEmpty() {
        return this.value.length == 0;
    }
```

I think that `length() == 0` and `isEmpty()` are equals in term of performance and readability of the codebase. If you want to improve the robustness of this check, what you could do is instead:

```suggestion
                if (sectionPart.isBlank()) {
```

This filters empty spaces too.

Reply:
`sectionPart` is StringBuilder which has no method `isBlank`.

Reply:
Then it's ok